### PR TITLE
Feature: static-server

### PR DESCRIPTION
**Pull request by jpettersson**
This is a new feature which improves dev server performance in request heavy contexts. For instance, when using the lingon server while running e2e tests with karma or similar. 

The Tunigo team ran into an issue where files would randomly be corrupted when requested from lingon in their Karma / phantomjs setup. This was probably caused by a combination of them having many tests as well as heavy processing on source files.

The new `lingon static-server` command will first do a full build and then serve static files directly from the `build` directory instead of building them on every request.

While working on this I also took the opportunity to refactor the dev server to be more modular.

@philipvonbargen

### CODE REVIEW

File: lib/server/buildCallback.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+var send              = require('send');
```
Comment (philipvonbargen) [inline lib/server/buildCallback.js]: Can you add `'use strict';` to the top?

File: lib/server/dynamicRequestHandler.js [hunk 2]
```diff
@@ -1,32 +1,11 @@
-'use strict';
```
Comment (philipvonbargen) [inline lib/server/dynamicRequestHandler.js]: Can you leave `'use strict';` in?

**Comment by philipvonbargen** [comment]
:+1: besides the two comments

**Comment by jpettersson** [comment]
Aaah, yes! I always forget that.

**Comment by jpettersson** [comment]
@pvb: The system tests in travis are very flaky these days.. seems they time out a lot

**Comment by PvB** [comment]
@jpettersson I guess you meant to write to @philipvonbargen :-)

**Comment by jpettersson** [comment]
@pvb: haha.. yes.. I'm pretty sure I've made this mistake before :) sorry!

**Pull request merged**